katdal/chunkstore.py

```diff
@@ -66,8 +81,12 @@ def generate_chunks(shape, dtype, max_chunk_size, dims_to_split=(0, 1)):
             # Split the dimension into the maximum number of chunks
             chunk_sizes = (1,) * shape[dim]
         else:
+            if power_of_two and _ceil_power_of_two(shape[dim]) == shape[dim]:
+                chunks_per_dim = _ceil_power_of_two(num_chunks)
+            else:
+                chunks_per_dim = np.ceil(num_chunks)
```

Comment:
Any particular reason to turn off the power-of-two feature if the shape isn't a power-of-two? For example, one might discard the first and last 1024 of 32768 channels, but still prefer to have a chunk size of 128 (which will divide in evenly) than 125.

Reply:
You've already written the code. Just remove the `if` statement and the `else` branch. There may be cases where you have only mostly power-of-two chunks plus a small piece at the end, but is that the end of the world?

Reply:
This is just a bit trickier than anticipated... For example, if we have 32768 channels for starters and we lob off 1024 at each end we get 30720 = 15 * 2048. So it is possible to get a nice power-of-2 compliant split if chunk_size = 2048. But what if the desired chunksize is 8192? I've now implemented this as 30720 = 8192 + 8192 + 8192 + 6144, giving more importance to the target object size than the power-of-two requirement (better to get most chunks large enough, I think).

I've thought of moving the power-to-two business to vis_writer but I prefer centralised logic to get the same splits everywhere.